chia/rpc/farmer_rpc_api.py

```diff
@@ -159,6 +218,45 @@ async def get_harvesters(self, _: Dict):
     async def get_harvesters_summary(self, _: Dict[str, object]) -> Dict[str, object]:
         return await self.service.get_harvesters(True)
 
+    async def get_harvester_plots_valid(self, request_dict: Dict[str, object]) -> Dict[str, object]:
+        # TODO: Consider having a extra List[PlotInfo] in Receiver to avoid rebuilding the list for each call
+        request = dataclass_from_dict(PlotInfoRequestData, request_dict)
+        plot_list = list(self.service.get_receiver(request.peer_id).plots().values())
+        # Apply filter
+        plot_list = [
+            plot for plot in plot_list if all(is_filter_match(plot, filter_item) for filter_item in request.filter)
+        ]
+        restricted_sort_keys: List[str] = ["pool_contract_puzzle_hash", "pool_public_key", "plot_public_key"]
+        # Apply sort_key and reverse if sort_key is not restricted
+        if request.sort_key in restricted_sort_keys:
+            raise KeyError(f"Can't sort by optional attributes: {restricted_sort_keys}")
+        # Sort by plot_id also by default since its unique
+        plot_list = sorted(plot_list, key=operator.attrgetter(request.sort_key, "plot_id"), reverse=request.reverse)
+        return paginated_plot_request(plot_list, request)
+
+    def paginated_plot_path_request(
+        self, source_func: Callable[[Receiver], List[str]], request_dict: Dict[str, object]
+    ) -> Dict[str, object]:
+        request: PlotPathRequestData = dataclass_from_dict(PlotPathRequestData, request_dict)
+        receiver = self.service.get_receiver(request.peer_id)
+        source = source_func(receiver)
+        request = dataclass_from_dict(PlotPathRequestData, request_dict)
+        # Apply filter
+        for filter_item in request.filter:
+            source = [plot for plot in source if filter_item in plot]
```

Comment:
To explain, my point, that may not be the important one, is that this form avoids creating several possibly large intermediate lists.

```suggestion
        source = [plot for plot in source if all(filter_item in plot for filter_item in request.filter)]
```